test/client/test_admin.py

```diff
@@ -0,0 +1,126 @@
+import re
+from pathlib import Path
+from tempfile import NamedTemporaryFile
+from textwrap import dedent
+
+import pytest
+from matchbox.admin import load_datasets_from_config
+from matchbox.common.db import Source, SourceWarehouse
+from tomli_w import dumps
+
+
+def warehouse_toml(warehouse: SourceWarehouse) -> str:
+    return dedent(f"""
+        [warehouses.{warehouse.alias}]
+        db_type = "{warehouse.db_type}"
+        user = "{warehouse.user}"
+        password = "{warehouse.password}"
+        host = "{warehouse.host}"
+        port = {warehouse.port}
+        database = "{warehouse.database}"
+    """).strip()
+
+
+def source_toml(source: Source, index: list[dict[str, str]]) -> str:
+    index_str = dumps({"index": index}).replace("\n", "\n        ")
+    return dedent(f"""
+        [datasets.{re.sub(r"[^a-zA-Z0-9]", "", source.alias)}]
+        database = "test_warehouse"
+        db_schema = "{source.db_schema}"
+        db_table = "{source.db_table}"
+        db_pk = "{source.db_pk}"
+        {index_str}
+    """)
+
+
+@pytest.mark.parametrize(
+    "index",
+    (
+        [
+            {"literal": "company_name"},
+            {"literal": "crn"},
+        ],
+        [{"literal": "company_name", "type": "VARCHAR", "alias": "name"}],
+        [
+            {"literal": "company_name"},
+            {"literal": "*"},
+        ],
+        [
+            {"literal": "*"},
+            {"literal": "company_name"},
+        ],
+    ),
+    ids=["vanilla", "alias_and_type", "star_end", "star_start"],
+)
+def test_load_datasets_from_config(
+    index: list[dict[str, str]],
+    warehouse: SourceWarehouse,
+    warehouse_data: list[Source],
+):
+    """Tests loading datasets from a TOML file."""
+    # Construct TOML from CRN data
+    # Columns: "id", "company_name", "crn"
+    crn = warehouse_data[0]
+    raw_toml = dedent(f"""
+        {warehouse_toml(warehouse)}
+        {source_toml(crn, index)}      
+    """).strip()
+
+    with NamedTemporaryFile(suffix=".toml", delete=False) as temp_file:
+        temp_file.write(raw_toml.encode())
+        temp_file.flush()
+        temp_file_path = Path(temp_file.name)
+
+    # Ingest
+    config = load_datasets_from_config(temp_file_path)
+
+    # Helper variables
+    source = config.get(re.sub(r"[^a-zA-Z0-9]", "", crn.alias))
```

Comment:
so aliases can't have underscores?

Reply:
Still unclear. Why would there be a dot in aliases? Are we happy with, e.g. a curly brace?